MyCapytain/resolvers/capitains/local.py

```diff
@@ -312,47 +287,38 @@ def parse(self, resource):
                 invalids.append(text)
 
         # Dispatching routine
-        for textgroup in collections:
-            self._dispatch_container(textgroup, os.path.dirname(textgroup.path))
+        for collection in collections:
+            self._dispatch_container(collection, os.path.dirname(collection.path))
 
         # Clean invalids if there was a need
         self._clean_invalids(invalids)
 
         self.inventory = self.dispatcher.collection
         return self.inventory
 
-    def __getText__(self, urn):
-        """ Returns a CtsTextMetadata object
-        :param urn: URN of a text to retrieve
-        :type urn: str, URN
+    def __getText__(self, identifier):
+        """ Returns a XmlCapitainsReadableMetadata object
+        :param identifier: URN of a text to retrieve
+        :type identifier: str, URN
         :return: Textual resource and metadata
-        :rtype: (CapitainsCtsText, InventoryText)
+        :rtype: (CapitainsCtsText, XmlCapitainsReadableMetadata)
         """
-        if not isinstance(urn, URN):
-            urn = URN(urn)
-        if len(urn) != 5:
-            if len(urn) == 4:
-                urn, reference = urn.upTo(URN.WORK), str(urn.reference)
-                urn = [
-                    t.id
-                    for t in self.texts
-                    if t.id.startswith(str(urn)) and isinstance(t, CtsEditionMetadata)
-                ]
-                if len(urn) > 0:
-                    urn = URN(urn[0])
-                else:
-                    raise UnknownObjectError
-            else:
-                raise InvalidURN
-
-        text = self.inventory[str(urn)]
+        # This will raise an UnknownCollection error if the identifier does not exist in the collection
+        # I assume that this is the desired outcome. If not, we can do a try/except here.
+        temp_text = self.inventory[str(identifier)]
+        if temp_text.readable:
+            text = temp_text
+        else:
+            # Perhaps there is a better exception to raise here?
+            raise(UnknownObjectError('{} is not a readable object'.format(str(identifier))))
 
         if os.path.isfile(text.path):
             with io.open(text.path) as __xml__:
-                resource = self.classes["text"](urn=urn, resource=self.xmlparse(__xml__))
+                resource = self.classes["text"](urn=identifier, resource=self.xmlparse(__xml__))
```

Comment:
most probably we should drop URN as a parameter here.

Reply:
Do you mean simply rename it (to e.g., `identifier`)? Or do mean to completely get rid of it?

Reply:
Definitely moving to identifier :)